x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -127,12 +131,15 @@ PERCENT : '%';
 OPENING_BRACKET : '[' -> pushMode(EXPRESSION), pushMode(EXPRESSION);
 CLOSING_BRACKET : ']' -> popMode, popMode;
 
+fragment UNQUOTED_ID_BODY
+    : (LETTER | DIGIT | DOT | UNDERSCORE)
+    ;
 
 UNQUOTED_IDENTIFIER
-    : LETTER (LETTER | DIGIT | '_')*
+    : LETTER UNQUOTED_ID_BODY*
```

Comment:
This will allow digits inside "subfields", but not on the root field. For example `123elasticsearch.node.stats.os.cpu.load_avg.1m`

Reply:
Do we want to allow unquoted identifiers that start with digits, though? If so, we'd at least have to disallow identifiers being all digits, no?

Reply:
> we'd at least have to disallow identifiers being all digits, no?

ES accepts both fields `"123"` and `"123.456"`.

Reply:
Which we accept as long as the fields are quoted "123" - same with field above "a.1m.4321" is accepted; the problem is handling fields which are NOT quoted.